docs/backends/gcloud.rst

```diff
@@ -53,7 +53,16 @@ In most cases, the default service accounts are not sufficient to read/write and
 #. Make sure your service account has access to the bucket and appropriate permissions. (`Using IAM Permissions <https://cloud.google.com/storage/docs/access-control/using-iam-permissions>`__)
 #. Ensure this service account is associated to the type of compute being used (Google Compute Engine (GCE), Google Kubernetes Engine (GKE), Google Cloud Run (GCR), etc)
 
-For development use cases, or other instances outside Google infrastructure:
+**Note:** There is currently a limitation in the GCS client for Python which by default requires a service account private key file to be
+present when generating signed urls. The service account private key is unavailable when running on a compute service.
+Compute Services (App Engine, Cloud Run, Cloud Functions, Compute Engine...) fetch `access tokens from the metadata server <https://cloud.google.com/docs/authentication/application-default-credentials>`__ .
+These services do not have access to the service account private key. This means that when trying to sign data in these services,
+you **MUST** use Cloud IAM sign function (SignBlob) to sign data and directly signing data isn't possible by any means.
+
+Luckily this can be worked around by passing `service_account_email` and `access_token` to the generate_signed_url function.
+When both of those args are provided, generate_signed_url will use the IAM SignBlob API to sign the url and no private key file is needed.
```

Comment:
Can you specify which settings need to be set for the storage to use this path.

Reply:
Just did